packages/core/src/i18n/i18n-setup.js

```diff
@@ -0,0 +1,41 @@
+import i18next from 'i18next';
+import LanguageDetector from 'i18next-browser-languagedetector';
+
+import mainTag from './language-detector';
+import enTranslation from './translations/en';
+import esTranslation from './translations/es';
+
+const languageDetector = new LanguageDetector();
+languageDetector.addDetector(mainTag);
+i18next.use(languageDetector).init({
+  fallbackLng: 'en',
+  detection: {
+    order: ['mainTag', 'htmlTag'],
+  },
+  resources: {
+    en: { translation: enTranslation },
+    es: { translation: esTranslation },
+  },
+});
```

Comment:
This initialization will detect the language on a page load - language detection doesn't automatically happen when things _in_ the DOM change.